src/components/organisms/UiSidePanel/UiSidePanel.vue

```diff
@@ -161,176 +161,163 @@ import {
   /**
    * Use this props to pass attrs for UiBackdrop.
    */
-  backdropAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  backdropAttrs?: BackdropAttrsProps;
   /**
    * Use this props to pass attrs for dialog element
    */
-  dialogAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  dialogAttrs?: DefineAttrsProps<null, DialogHTMLAttributes>;
   /**
    * Use this props to pass attrs for dialog Transition.
    */
-  transitionDialogAttrs: {
-    type: Object,
-    default: () => ({
-      appear: true,
-      name: 'slide',
-    }),
-  },
+  transitionDialogAttrs?: TransitionProps;
   /**
    * Use this props to pass attrs for title UiHeading.
    */
-  headingTitleAttrs: {
-    type: Object as PropsAttrs,
-    default: () => ({ level: 2 }),
-  },
+  headingTitleAttrs?: HeadingAttrsProps;
   /**
    * Use this props to pass attrs for subtitle UiText.
    */
-  textSubtitleAttrs: {
-    type: Object as PropsAttrs,
-    default: () => ({}),
-  },
+  textSubtitleAttrs?: TextAttrsProps;
   /**
    * Use this props to pass attrs for close UiButton
    */
-  buttonCloseAttrs: {
-    type: Object as PropsAttrs,
-    default: () => ({}),
-  },
+  buttonCloseAttrs?: ButtonAttrsProps;
   /**
    * Use this props to pass attrs for close UiIcon
    */
-  iconCloseAttrs: {
-    type: Object,
-    default: () => ({ icon: 'close' }),
-  },
+  iconCloseAttrs?: IconAttrsProps;
   /**
    * Use this props to pass attrs for content element
    */
-  contentAttrs: {
-    type: Object,
-    default: () => ({}),
-  },
+  contentAttrs?: DefineAttrsProps<null>;
+}
+export type SidePanelAttrsProps = DefineAttrsProps<SidePanelProps>;
+export interface SidePanelEmits {
+  (e: 'update:modelValue', value: boolean): void;
+  (e: 'after-enter'): void;
+}
+
+const props = withDefaults(defineProps<SidePanelProps>(), {
+  modelValue: false,
+  title: '',
+  subtitle: '',
+  transitionBackdropAttrs: () => ({
+    appear: true,
+    name: 'fade',
+  }),
+  backdropAttrs: () => ({}),
+  dialogAttrs: () => ({}),
+  transitionDialogAttrs: () => ({
+    appear: true,
+    name: 'slide',
+  }),
+  headingTitleAttrs: () => ({ level: 2 }),
+  textSubtitleAttrs: () => ({}),
+  buttonCloseAttrs: () => ({}),
+  iconCloseAttrs: () => ({ icon: 'close' }),
+  contentAttrs: () => ({}),
 });
-const emit = defineEmits<{(e: 'update:modelValue', value: boolean): void,
-  (e: 'after-enter'): void
-}>();
+const emit = defineEmits<SidePanelEmits>();
 const button = ref<InstanceType<typeof UiButton>| null>(null);
-async function enterHandler(): Promise<void> {
+const enterHandler = async () => {
   await nextTick();
   focusElement(button.value?.$el, true);
-}
-function afterEnterHandler(): void {
+};
+const afterEnterHandler = () => {
   emit('after-enter');
-}
+};
 // TODO: remove in 0.6.0 / BEGIN
 const attrs = useAttrs();
-const transition = computed(() => attrs.transition);
+const transition = computed(() => attrs.transition as string | undefined);
```

Comment:
I'm not convinced here... Is this computed property really any possible string or an undefined? Or can we specify it somehow? 🤔 😄

Reply:
yeah, you're right. I can use `SidePanelProps['transitionDialogAttrs']`, thanks 😀